### ISSUE TITLE
Consensus grandpa finality lag

### ISSUE THREAD

**Issue by flipchan** [liberland/liberland_substrate#144]
backoff_authoring_blocks = None

**Comment by flipchan** [comment]
```
2022-11-20 06:56:25 Detected prevote equivocation in the finality worker: Equivocation { round_number:
 1, identity: Public(898d312c59c405816d2435f2fe8141f051be1ee7285418ac738d3886911f0332 (5FB4Mqt2...)),
first: (Prevote { target_hash: 0x2dd57e74fe3aee6a811a4fa466d908edf62b5a34a207d95d4ad084bdd8551040, tar
get_number: 17955 }, Signature(88b405685c19e71d2a6d579337033aabf67670ea91354fb766f017d580a5c59793aeef5
6c56054a1aa2220607f0c2dcfaafa175c8bcbd94907ccc46efb340608)), second: (Prevote { target_hash: 0x4806c71
79f354aaf345ecc8042a7ac7c7feeff51fe413b27a26937a93be7d266, target_number: 18071 }, Signature(a5d213833
4fafce0c4525dde8117559bc5ab58c749241a083c49e5e531ca31015a96c543a8d0728b62619895f4addcff18154b658ce0c68
75724d5b7dbd3fb06)) }
2022-11-20 06:56:25 Error reporting prevote equivocation: safety invariant has been violated: Refraini
ng from sending equivocation report for our own equivocation.                                        
2022-11-20 06:56:27 💤 Idle (3 peers), best: #18192 (0xdd9c…b2d1), finalized #17955 (0x2dd5…1040), ⬇ 0
.4kiB/s ⬆ 0.7kiB/s

```

**Comment by flipchan** [comment]
```
$ polkadot-js-api --ws ws://computer:9944 tx.grandpa.noteStalled 3 17955 --sudo --seed ""
{
  "noteStalled": {
    "events": [],
    "status": "Ready"
  }
}
{
  "noteStalled": {
    "events": [],
    "status": {
      "Broadcast": [
        "12D3KooWDpE9vnnL857xBzcP24nTiMWMsZ9h2hBMfbkR9eSv7473",
        "12D3KooWC4cHgjNDNuwmd5nsppZ54QKNRRqqkx66uvLcjxtr2Jar",
        "12D3KooWGWY6RWHm4b6w6FkWsSSNvxJvbCNnP4oPzkL45qmRo2Jp"
      ]
    }
  }
}


```

**Comment by flipchan** [comment]
GRANDPA voter error: could not complete a round on disk: Database 
                                                                  
Essential task `grandpa-voter` failed. Shutting down service.

**Comment by kacperzuk-neti** [comment]
@flipchan I'd like to try to reproduce this. Did you find anything specific needed to reproduce it? Is it enough to setup local chain with multiple validatiors and wait for validator set change?

**Comment by flipchan** [comment]
@kacperzuk-neti yeah, you need to setup a local instance with 3 validators and then add a 4th one

**Comment by flipchan** [comment]
fixed in #152
